java-checks/src/main/java/org/sonar/java/syntaxtoken/FirstSyntaxTokenFinder.java

```diff
@@ -294,4 +321,112 @@ public void visitNewClass(NewClassTree tree) {
   public void visitParenthesized(ParenthesizedTree tree) {
     firstSyntaxToken = tree.openParenToken();
   }
+
+  @Override
+  public void visitBinaryExpression(BinaryExpressionTree tree) {
+    scan(tree.leftOperand());
+  }
+
+  @Override
+  public void visitCompilationUnit(CompilationUnitTree tree) {
+    if (tree.packageName() != null) {
+      // TODO(SONARJAVA-547) Should be the package token
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1135)